### ISSUE TITLE
False positive for `call_user_func`

### ISSUE THREAD

**Issue by chriscpty** [kalessil/phpinspectionsea#1965]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended)  |
| Language level | PHP 8.1                        |

Current behaviour
--
I have a function like this:

```php
public function find(array $arr, \Closure|string $callback)
{
  foreach ($arr as $val) {
    if (call_user_func($callback, $val)) {
      return $val;
    }
  }
  return null;
}
```

This shows a lint saying that "[EA] '$callback($contact)' would make more sense here (it also faster).".

Expected behaviour
--
As `$callback` isn't necessarily a `Closure`, simply calling it as `$callback($contact)` fails when it is a function name, as far as I am aware.
The lint is thus incorrect, it should only show up if `$callback` is guaranteed (by argument typing, PHPDocs or because we know the scope it is created in) to be a `Closure`.

**Comment by jdreesen** [comment]
> As $callback isn't necessarily a Closure, simply calling it as $callback($contact) fails when it is a function name, as far as I am aware.

I don't think this is true, see: https://3v4l.org/T0Mjq

**Comment by chriscpty** [comment]
TIL, I thought that didn't work, my bad.
